junit5/src/main/java/com/linecorp/armeria/testing/server/ServiceRequestContextCaptor.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.testing.server;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.SimpleDecoratingHttpService;
+
+/**
+ * Captures the {@link ServiceRequestContext}s.
+ * <pre>{@code
+ * class ServiceRequestContextCaptorTest {
+ *   static final ServiceRequestContextCaptor captor = new ServiceRequestContextCaptor();
+ *
+ *   @RegisterExtension
+ *   static final ServerExtension server = new ServerExtension() {
+ *       @Override
+ *       protected void configure(ServerBuilder sb) throws Exception {
+ *           sb.service("/hello", (ctx, req) -> HttpResponse.of(200));
+ *           sb.decorator(captor.decorator());
+ *       }
```

Comment:
This is not correct anymore.

Reply:
> This is not correct anymore.

Do you mean `captor.decorator()`? It is still available, we just add a variant accepting `filter`, no?

Reply:
If a user uses a `ServerException`, it will capture the contexts already, so it doesn't make much sense to capture again. How about replacing this example with two examples?

1. The example that doesn't use `ServerExtension` but uses the captor manually.
2. The example that uses `ServerExtension`.

Reply:
Done